aips/aip-x.md

```diff
@@ -0,0 +1,158 @@
+---
+aip: x
+title: Aptos Token Objects Framework Update
+author: johnchanguk
+discussions-to (*optional):
+Status: In Review
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Framework
+created: 03/23/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-X - Aptos Token Objects Framework Update
+
+## Summary
+
+This AIP aims to extend the `collection`, `token` and `property_map` modules from the `aptos-token-objects` framework. Reasons for this are to introduce extensibility and customization to collection and token objects.
+
+## Motivation
+
+Currently there is no way of changing the name or max supply of a collection, nor create a token with the token and custom seed provided. Reasons for wanting this is for example:
+
+1. Some users may not want to have on chain data publicly visible, such as the collection name during collection creation. Being able to set this at a later date allows users to confidently deploy collections without leaking any sensitive information.
+2. Users may want to adjust the max supply for a collection. This can occur if there needs to be adjustments in total supply, or simply rectifying mistakes.
+3. If a user wants to predict the address of a token before creating this on chain, this AIP would allow this by taking in the token and custom seed. From this, the address of the token can be known beforehand. An example of why this is needed is preparing metadata for tokens that are yet to be minted.
```

Comment:
`create_named_token` already provides this, what are we gaining by adding the additional here?

the motivating example is already solved with the current standard.

Reply:
What if I want the same token name for all tokens in a collection, as well as predictable addresses? The current implementation would fail as it takes the `name` as the seed. Users may want this option.

Reply:
add it as a motivating example :P

Reply:
Added